parser_library/src/processing/preprocessor.cpp

```diff
@@ -63,21 +63,20 @@ std::vector<std::shared_ptr<semantics::preprocessor_statement_si>> preprocessor:
 }
 
 void preprocessor::do_highlighting(
-    const semantics::preprocessor_statement_si& stmt, semantics::source_info_processor& src_proc)
+    const semantics::preprocessor_statement_si& stmt, semantics::source_info_processor& src_proc) const
 {
-    if (stmt.label_ref().type != semantics::label_si_type::EMPTY)
-        src_proc.add_hl_symbol(token_info(stmt.label_ref().field_range, semantics::hl_scopes::label));
+    const auto& details = stmt.m_details;
 
-    src_proc.add_hl_symbol(token_info(stmt.instruction_ref().field_range, semantics::hl_scopes::instruction));
+    if (!details.label.name.empty())
+        src_proc.add_hl_symbol(token_info(details.label.r, semantics::hl_scopes::label));
 
-    for (const auto& op : stmt.operands_ref().value)
-    {
-        if (op)
-            src_proc.add_hl_symbol(token_info(op->operand_range, semantics::hl_scopes::operand));
-    }
+    src_proc.add_hl_symbol(token_info(details.instruction.r, semantics::hl_scopes::instruction));
+
+    if (!details.operands.first.empty())
+        src_proc.add_hl_symbol(token_info(details.operands.second, semantics::hl_scopes::operand));
```

Comment:
does this work correctly with multiline statements in file that have sequence number in the columns 73+?